{README.md, docker-compose.yml, services/*}: Add tutorial to README, fix docker-compose, fix and improve services (#5)
DataDog/go-sample-app: README.md

```sh
$ orchestrion -w .
Scanning Package /.../go-sample-app
overwriting /.../go-sample-app/services/notes/main.go:
overwriting /.../go-sample-app/services/users/main.go:
overwriting /.../go-sample-app/tools/header_check.go:
```

3. If you are curious about the changes orchestrion made, review the git diff:
```sh
$ git diff
...
```

You should see new function calls and wrappers placed around relevant code such as HTTP handler functions and SQL clients. You'll also see comments added, such as `//dd:startwrap`. These comments are used to identify instrumented code, so that Orchestrion can be run multiple times without duplicating instrumentation, and also so Orchestrion can remove instrumentation when instructed to with the `-rm` flag.

4. Shut down the services:
```
$ docker-compose down
```

5. You need to update the `go.mod` files before you can build the new code. The Orchestrion instrumentation requires code from the `github.com/DataDog/orchestrion` library:

```shell
$ cd services/users
$ go mod tidy
$ cd ../notes/
$ go mod tidy
$ cd ../../
```

6. Rebuild the applications with the newly instrumented code:
```shell
$ docker-compose build
```

> 7. And start the services again:
> | :zap: Remember that the DD_API_KEY environment variable needs to be set! |
> |--------------------------------------------------------------------------|
> ```shell

Reviewer: ```suggestion
7. And start the services again:

| :zap: Remember that the DD_API_KEY environment variable needs to be set! |
|--------------------------------------------------------------------------|

```shell
```
Add newlines so this renders correctly.

A sample application composed of several services, written in Go.

This application is a downloadable, interactive example that demonstrates the use of Datadog's Go auto-instrumentation ([DataDog/orchestrion](https://github.com/DataDog/orchestrion))

> 

Reviewer 2: Looks complete. Thanks! You might want Kari or Brett to give a pass on the wording. We also might want to make this a page on the DD public docs site in addition to being in the README.

Reviewer: Sounds good, Jon! I talked to Kyle about single-sourcing this eventually. I'll set it up so the README connects to a page on the public docs title.

Kyle mentioned that we should wait to do this until this GAs at the end of the quarter. That'll be my plan unless anything changes. :)

go: found github.com/datadog/orchestrion/instrument in github.com/datadog/orchestrion v0.1.0
services/users $ cd ../notes/
services/notes $ go mod tidy
go: finding module for package github.com/datadog/orchestrion/instrument
go: found github.com/datadog/orchestrion/instrument in github.com/datadog/orchestrion v0.1.0
services/notes $ cd ../../
$ 
```

Then rebuild the applications with our newly instrumented code
```sh 
$ docker-compose build
```

And start the services again:
| :zap: Remember that the DD_API_KEY environment variable needs to be set! |
|--------------------------------------------------------------------------|
```sh 
$ docker-compose up -d
```

Now, going to the application home page [http://localhost:8080](http://localhost:8080) will show us the same page, but looking at our Traces Page in the Datadog app [https://app.datadoghq.com/apm/traces](https://app.datadoghq.com/apm/traces) should show some new traces appear!

![First Trace](doc/trace1.png)

Click through the various spans and tabs. For instance, by clicking the `sqlite3.query` span, we can see the SQL query and other info.
![SQLite3 Trace Data](doc/sqlite-info.png)

> ## More Traces
> Now that we have instrumentation, let's generate some more interesting traces. From the browser, click on one of the users' IDs to go to their notes page.

Reviewer: ```suggestion
## Generating more traces

You've added some basic instrumentation to the sample application, so you can now generate some more interesting traces.

1. From the sample application in your browser, click one of the users' IDs to go to their notes page.
```

Navigating to that page should generate another trace, which looks like this:
![Notes Page](doc/notes-trace.png)

> Here we can see that the `users` service received an http request. It then sent an outgoing http request to the `notes` service, requesting `/notes?userid=2` in this case (this will depend on which user's page you are viewing).

Reviewer: ```suggestion
Here, you can see that the `users` service received an HTTP request. It then sent an outgoing HTTP request to the `notes` service, requesting `/notes?userid=2`. This varies depending on which user's page you view.
```

> ![Outgoing](doc/notes-trace2.png)

Reviewer: ```suggestion

![Outgoing](doc/notes-trace2.png)
```
Help this image wrap to a newline.

The `notes` service then executes a SQL query:
![Notes SQL Query](doc/notes-sql.png)

> 
> Try adding a new note. When you do that, you should see a new trace appear.
> ![New Note Trace](doc/notes-trace3.png)
> This trace appears with type `POST` and status code `302` because the `users` service returns a redirect to the browser after the submission succeeds.

Reviewer: ```suggestion
2. Add a new note. When you do, you should see a new trace appear.

![New Note Trace](doc/notes-trace3.png)

This trace appears with type `POST` and status code `302` because the `users` service returns a redirect to the browser after the submission succeeds.
```

> Once again, we can see in the trace that the `users` service talks to the `notes` service, this time resulting in an `INSERT` query.

Reviewer: ```suggestion
3. Examine the trace again. You can see that the `users` service communicates with the `notes` service, this time resulting in an `INSERT` query.
```

![New Note Trace Details](doc/notes-trace4.png)

> ## Removing Instrumentation
> Now that we've seen how to add instrumentation to the code, we might as well cover how to remove it. 
> 
> Removing instrumentation is as easy as adding it. If, for whatever reason, you want to remove the instrumentation added by `orchestrion`, run the following in the repository root:

Reviewer: ```suggestion
## Removing Instrumentation

Removing instrumentation is as simple as adding it. To remove the instrumentation added by Orchestrion:

1. Run the following in the repository root:
```

> ```sh 
> $ orchestrion -rm -w .
> Scanning Package /.../go-sample-app
> Removing Orchestrion instrumentation.
> overwriting /.../go-sample-app/services/notes/main.go:
> overwriting /.../go-sample-app/services/users/main.go:
> overwriting /.../go-sample-app/tools/header_check.go:
> ```

Reviewer: ````suggestion
```shell
$ orchestrion -rm -w .
```
````
Similar to the one of the previous suggestions, it think it's simpler to only list the commands a user needs to execute.

> Next, run `go mod tidy` again, as we did before, for both services. This will clean up the dependencies we had to add earlier.

Reviewer: ```suggestion
2. Run `go mod tidy` again, as you did before, for both services. This cleans up the dependencies from before.
```

> A git diff should show that the instrumentation has been removed.

Reviewer: ```suggestion
3. Run `git diff` to verify you removed the instrumentation:
```
